pkg/api/types.go

```diff
@@ -209,6 +209,8 @@ type CustomNodesDNS struct {
 type WindowsProfile struct {
 	AdminUsername             string            `json:"adminUsername"`
 	AdminPassword             string            `json:"adminPassword" conform:"redact"`
+	CsiProxyURL               string            `json:"csiProxyUrl"`
+	EnableCsiProxy            *bool             `json:"enableCsiProxy,omitempty"`
```

Comment:
nit: s/EnableCsiProxy/EnableCSIProxy and s/enableCsiProxy/enableCSIProxy

I know